pegasus/helpers/section_api_helpers.rb

```diff
@@ -21,23 +21,32 @@ def self.fetch_user_students(user_id)
   def self.create(params)
     name = !params[:name].to_s.empty? ? params[:name].to_s : 'New Student'
     gender = valid_gender?(params[:gender]) ? params[:gender] : nil
+    provider = supported_provider?(params[:provider]) ? params[:provider] : 'sponsored'
     birthday = age_to_birthday(params[:age]) ?
       age_to_birthday(params[:age]) : params[:birthday]
 
     created_at = DateTime.now
 
-    row = Dashboard.db[:users].insert(
+    data =
       {
         name: name,
         user_type: 'student',
-        provider: 'sponsored',
+        provider: provider,
         gender: gender,
         birthday: birthday,
         created_at: created_at,
         updated_at: created_at,
         username: UserHelpers.generate_username(Dashboard.db[:users], name)
       }.merge(random_secrets)
-    )
+    if provider == 'sponsored'
+      row = Dashboard.db[:users].insert(data)
+    else
+      uid = params[:uid].to_s
+      data[:uid] = uid
+      row = Dashboard.db[:users].find_or_create(provider: provider, uid: uid) do |user|
+        user.merge!(data)
```

Comment:
Am I mistaken in thinking that `find_or_create_by` is an [ActiveRecord method](https://apidock.com/rails/v4.0.2/ActiveRecord/Relation/find_or_create_by) not offered on [Sequel::Dataset](http://www.rubydoc.info/github/jeremyevans/sequel/Sequel/Dataset)?

If it were to exist, I think this would erroneously overwrite many fields that we wouldn't want to update, e.g., I don't think we would want to update `created_at`, `birthday`, or `secret_picture_id` (for example).